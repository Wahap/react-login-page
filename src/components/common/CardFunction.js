import Card from "react-bootstrap/Card";
import CardDeck from "react-bootstrap/CardDeck";
import React from "react";
const CardFunction = props => {
  return (
    <>
      <div>
        {props.data.map((post, i) => {
          return (
            <div key={i}>
              <CardDeck>
                <Card bg="primary" text="white" style={{ width: "18rem" }}>
                  <Card.Header>{post.title} </Card.Header>
                  <Card.Body>
                    <Card.Title>Primary Card Title</Card.Title>
                    <Card.Text>{post.body}</Card.Text>
                    <button
                      style={{ marginBottom: 20 }}
                      className="btn btn-primary add-course"
                      onClick={() => props.onClick()}
                    >
                      stop
                    </button>
                  </Card.Body>
                </Card>
              </CardDeck>
              <br />
            </div>
          );
        })}
      </div>
    </>
  );
};

export default CardFunction;
