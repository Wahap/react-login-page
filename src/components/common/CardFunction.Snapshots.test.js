import React from "react";
import PropTypes from "prop-types";
import renderer from "react-test-renderer";
import { posts, postArray } from "../../../tools/mockData";
import configureStore from "../../redux/configureStore";
import CardFunction from "./CardFunction";
const store = configureStore();

const props = {
  loading: true,
  posts
};

it("sets submit button label 'Saving...' when saving is false", () => {
  const tree = renderer.create(<CardFunction store={store} data={posts} />);

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(<CardFunction store={store} data={postArray} />);

  expect(tree).toMatchSnapshot();
});
