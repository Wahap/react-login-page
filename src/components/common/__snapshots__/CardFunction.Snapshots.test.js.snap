// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Saving...' when saving is false 1`] = `
<div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Securing React Apps with Auth0
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          React: The Big Picture
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Creating Reusable React Components
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Building a JavaScript Development Environment
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Building Applications with React and Redux
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Building Applications in React and Flux
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Clean Code: Writing Code for Humans
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Architecting Applications for the Real World
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Becoming an Outlier: Reprogramming the Developer Mind
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Web Component Fundamentals
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
</div>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<div>
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card bg-primary text-white"
        style={
          Object {
            "width": "18rem",
          }
        }
      >
        <div
          className="card-header"
        >
          Wahap
           
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Primary Card Title
          </div>
          <p
            className="card-text"
          />
          <button
            className="btn btn-primary add-course"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            stop
          </button>
        </div>
      </div>
    </div>
    <br />
  </div>
</div>
`;
